lib/services/elasticsearch.js

```diff
@@ -746,34 +746,45 @@ class ElasticSearch extends Service {
    *
    * @returns {Promise}
    */
-  createIndex (index) {
+  async createIndex (index) {
     this._assertValidIndexAndCollection(index);
 
-    return this._client.cat.indices({ format: 'json' })
-      .then(({ body }) => {
-        const esIndexes = body.map(({ index: name }) => name);
+    let body;
 
-        for (const esIndex of esIndexes) {
-          const indexName = this._extractIndex(esIndex);
+    try {
+      ({ body } = await this._client.cat.indices({ format: 'json' }));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment of "{ body }" from this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AAssignmentWithinCondition)